consensus/state_test.go

```diff
@@ -208,22 +211,19 @@ func TestStateBadProposal(t *testing.T) {
 	startTestRound(cs1, height, round)
 
 	// wait for proposal
-	ensureNewProposal(proposalCh)
+	ensureNewProposal(proposalCh, height, round)
 
 	// wait for prevote
-	ensureNewVote(voteCh)
-
+	ensureVote(voteCh, height, round, types.VoteTypePrevote)
 	validatePrevote(t, cs1, round, vss[0], nil)
 
 	// add bad prevote from vs2 and wait for it
 	signAddVotes(cs1, types.VoteTypePrevote, propBlock.Hash(), propBlock.MakePartSet(partSize).Header(), vs2)
-	ensureNewVote(voteCh)
+	ensureVote(voteCh, height, round, types.VoteTypePrevote)
 
 	// wait for precommit
-	ensureNewVote(voteCh)
-
+	ensureVote(voteCh, height, round, types.VoteTypePrecommit)
```

Comment:
maybe these should just be `ensurePrevote` and `ensurePrecommit` and we can drop the last arg